packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,225 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps,  DOMRef, FocusEvents, FocusableRef, InputDOMProps, ValueBase} from '@react-types/shared';
+import {Radio, RadioGroup, RadioProps, Provider} from "react-aria-components"
+import {useDOMRef, useFocusableRef} from '@react-spectrum/utils';
+import {createContext, forwardRef, ReactNode, useContext, useRef} from 'react';
+import {focusRing, StyleProps, getAllowedOverrides} from './style-utils' with {type: 'macro'};
+import {style} from '../style/spectrum-theme' with {type: 'macro'};
+import {centerBaseline} from './CenterBaseline';
+import {IconContext} from './Icon';
+import {raw} from '../style/style-macro' with {type: 'macro'};
+
+
+interface SegmentedControlProps extends ValueBase<string|null, string>, InputDOMProps, FocusEvents, StyleProps, AriaLabelingProps {
+  children?: ReactNode,
+  /**
+   * The style of the track.
+   * 
+   * @default 'none'
+   */
+  trackStyle?: 'none' | 'track',
+  isDisabled?: boolean
+}
+interface ControlItemProps extends Omit<RadioProps, 'children'>, StyleProps {
+  children?: ReactNode
+}
+
+const segmentedControl = style({
+  fontFamily: 'sans',
+  fontSize: 'control',
+  fontWeight: 'normal',
+  display: 'flex',
+  minWidth: {
+    trackStyle: {
+       // this is an arbitrary value. there is no minWidth in the tokens but the size of the each segment is dependent on the width of the segmented control so without it, it looks strange
+      track: 240
+    }
+  }
+}, getAllowedOverrides())
+
+const controlItem = style({
+  ...focusRing(),
+  display: 'flex',
+  gap: 'text-to-visual',
+  color: {
+    default: {
+      trackStyle: {
+        none: 'gray-600',
+        track: 'gray-700'
+      }
+    },
+    isHovered: 'neutral-subdued',
+    isSelected: 'neutral',
+    isDisabled: 'disabled',
+    forcedColors: {
+      default: 'ButtonText',
+      isDisabled: 'GrayText',
+      isSelected: 'HighlightText'
+    },
+  },
+  backgroundColor: {
+    trackStyle: {
+      none: {
+        isSelected: {
+          default: 'gray-100',
+          isHovered: 'gray-200',
+          isDisabled: 'disabled'
+        },
+        forcedColors: {
+          isSelected: 'Highlight',
+          isDisabled: {
+            isSelected: 'GrayText'
+          }
+        }
+      }, 
+      track: {
+        isSelected: {
+          default: 'gray-25'
+        },
+        forcedColors: {
+          isSelected: 'Highlight',
+          isDisabled: {
+            isSelected: 'GrayText'
+          },
+        }
+      }
+    },
+  },
+  // the padding should be a little less for segmented controls that only contain icons but not sure of a good way to do that
```

Comment:
im not sure if this will work in this scenario because there's a hidden input so it looks like
```
<div>
  <span with the visually hidden input>
  <icon>
</div>
```

so it's not actually an only-child like the way it is in button for example

Reply:
tch... annoying
well, you could use callback refs to know if you have both an icon and text

we should be using the same requirements, either just a string, in which case you wrap it in a Text. Or users must pass an Icon and/or/not Text

Reply:
the only thing is that it throws an error if you try to do `{typeof props.children === 'string' ? <Text>{props.children}</Text> : props.children}` in Radio because in RAC we expect the slot "description" or "error message" with `<Text>`. 

i could just use a span instead of `<Text>`